datastore-api/app/core/kgs/connector.py

```diff
@@ -172,30 +175,43 @@ async def edges_in_out_msearch(self, kg_name, nids):
         index = f'{kg_name}{self.datastore_suffix}'
 
         body = []
-        for nid in nids:
-            body.append({'index': index})
-            body.append({
-                "query": {
-                    "bool": {
-                        "filter": {
-                            "bool" : {
-                                "should" : [
-                                    {"term" : { "in_id" : nid } },
-                                    {"term" : { "out_id" : nid } },
-                                ]
-                            }
+
+        body.append({'index': index})
+        body.append({
+            "query": {
+                "bool": {
+                    "filter": {
+                        "bool" : {
+                            "should" : [
+                                {"terms" : { "in_id" : nids } },
+                                {"terms" : { "out_id" : nids } },
+                            ]
                         }
                     }
-                },
-                "size": 10000
-            })
-    
-        response = await self.es.msearch(body=body)
+                }
+            },
+            "size": 10000
+        })
+
+        responses = await self.es.msearch(body=body)
 
         results = {}
-        for nid, response in zip(nids, response['responses']):
-            edges = {hit['_id']: dict(hit['_source'], **{'_id': hit['_id']}) for hit in response["hits"]["hits"]}
-            results[nid] = edges
+
+        edges = {hit['_id']: dict(hit['_source'], **{'_id': hit['_id']}) for hit in responses["responses"][0]["hits"]["hits"]}
+        for edge in edges.values():
+            if edge['in_id'] in nids:
+                try:
+                    results[edge['in_id']]
+                except:
+                    results[edge['in_id']] = list()
+                results[edge['in_id']].extend([edge])
+            else:
+                try:
+                    results[edge['out_id']]
+                except:
+                    results[edge['out_id']] = list()
+                results[edge['out_id']].extend([edge])
+        
         return results
```

Comment:
Please annotate the return type and explain it for this method (since it seems very complex and people would not understand what the type is otherwise)